backend/langate/network/models.py

```diff
@@ -97,3 +84,27 @@ def create_user_device(user: User, ip, name=None):
 
         device = UserDevice.objects.create(mac=mac, name=name, user=user, ip=ip, area=area)
         return device
+
+def generate_dev_name():
+    """
+        Generate a random device name based on a list of names
+    """
+    try:
+        with open("assets/misc/device_names.txt", "r", encoding="utf-8") as f:
+            lines = f.read().splitlines()
+            devices_count = Device.objects.count() + UserDevice.objects.count()
```

Comment:
A UserDevice is a Device by inheritance, by doing this, you are counting each UserDevice twice.

Reply:
Oh, right.